app/controllers/users_controller.rb

```diff
@@ -16,17 +24,58 @@ def create
     if @user.save
       log_in @user
       flash[:success] = t "message_register_success"
-      redirect_to new_user_path
+      redirect_back_or @user
     else
       flash.now[:error] = t "message_register_error"
       render :new
     end
   end
 
+  def edit
+    @user = find_user params[:id]
+  end
+
+  def update
+    @user = find_user params[:id]
+    if @user.update user_params
+      flash[:success] = t "profile_message"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    user = find_user params[:id]
+    if user&.destroy
+      flash[:success] = t "user_delete_s"
+    else
+      flash[:danger] = t "user_delete_e"
+    end
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password,
                                  :password_confirmation
   end
+
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t "login_message"
+    redirect_to login_url
+  end
+
+  def correct_user
+    @user = find_user params[:id]
+    redirect_to root_url unless current_user?(@user)
+  end
+
+  def find_user params
+    User.find_by(id: params)
```

Comment:
nên xử lý trường hợp ko tìm thấy user ở ngay method này luôn e, khỏi phải check lại ở các chỗ sử dụng sau này

Reply:
chị ơi nếu tìm không thấy user thì return về thẳng lun hay mình trả về một giá trị (nil chẳng hạn ) ạ

Reply:
nếu ko tìm thấy user chú có thể đẩy về page login hoặc register user và thêm thông báo vui lòng login hoặc user not found nhé